x-pack/plugins/fleet/server/services/agents/crud.ts

```diff
@@ -161,34 +184,51 @@ export async function countInactiveAgents(
   return res.body.hits.total.value;
 }
 
-export async function getAgent(esClient: ElasticsearchClient, agentId: string) {
+export async function getAgentById(esClient: ElasticsearchClient, agentId: string) {
+  const agentNotFoundError = new AgentNotFoundError(`Agent ${agentId} not found`);
```

Comment:
I found it weird to create an error each time we call this function, I would duplicate this were we throw the error

Reply:
I thought it was preferable to waste a few microseconds creating a value we might not use to ensure the exit/error values remained in sync.

Reply:
can we ensure we have the same error using a function like `createAgentNotFoundError` for example